docs/shrc.md

```diff
@@ -66,43 +145,141 @@ The [-Xshareclasses](xshareclasses.md) option is highly configurable, allowing y
 
-## Cache utilities
 
-Several utilities are provided for managing active caches, which are invoked by specifying `-Xshareclasses` suboptions. These utilities control the following types of operations:
+### The Java shared classes Helper API
+
+The Java Helper API classes can be found in the [`com.ibm.oti.shared`](api-overview.md#monitoring-and-management) package.
+
+Each class loader that wants to share classes must get a `SharedClassHelper` object from a `SharedClassHelperFactory`. The `SharedClassHelper`, when created, has a one to one relationship with the class loader. That is, it belongs to the class loader that requested it and can only store classes defined by that class loader. The `SharedClassHelper` gives the class loader a simple API for finding and storing classes in the class cache to which the VM is connected. If the class loader is garbage collected, its `SharedClassHelper` is also garbage collected.
+
+The following main functions are available from the `SharedClassHelper` API:
+
+- `findSharedClass`: Used to check whether a class is already in the cache before looking for the class on the filesystem.
+- `storeSharedClass`: Used to store a class in the cache.
+- `setSharingFilter`: A filter that can be used to decide which classes are found and stored in the cache. This filter can be applied to a particular package by implementing the `SharedClassFilter` interface. To apply a filter to all non-bootstrap class loaders that share classes, specify the `-Dcom.ibm.oti.shared.SharedClassGlobalFilterClass` system property on the command line.
+
+Each class loader that wants to share data must get a `SharedDataHelper` object from a `SharedDataHelperFactory`.
+A `SharedDataHelperFactory` provides an interface that can be used to create `SharedDataHelpers`, which are used for storing Java byte array data. A `SharedDataHelper` also has a one to one relationship with a class loader, although a class loader can exist without a `SharedDataHelper`.
+
+
+### The Java shared classes utility API
+
+The following APIs are available for obtaining information about a shared classes cache:
+
+- `com.ibm.oti.shared.SharedClassStatistics`: Obtains information about cache size, including free space, soft maximum limit, and the limits enforced for AOT and JIT data.
+- `com.ibm.oti.shared.SharedClassUtilities`: Obtains detailed information about a shared classes cache, including its size, name, type, and status.
+- `com.ibm.oti.shared.SharedClassCacheInfo`: Stores information about a shared classes cache and provides API methods to retrieve the information and remove caches. For example, you can use the `IterateSharedCaches` and `DestroySharedCache` [JVMTI extensions](interface_jvmti.md#iteratesharedcaches).
```

Comment:
This shouldn't contain "For example, ". You can use the jvmti extensions as an alternate to SharedClassCacheInfo.